src/server.cpp

```diff
@@ -328,6 +328,7 @@ Server::~Server()
 			kick_msg = m_shutdown_state.message;
 		}
 		if (kick_msg.empty()) {
+			reconnect = g_settings->getBool("ask_reconnect_on_shutdown");
```

Comment:
However this breaks some cases: If Lua doesn't set a shutdown message then its value for `reconnect` will always get overwritten.